### PR TITLE
combine functions behind to/from-c/tsv commands

### LINKED ISSUE #969: combine csv and tsv functions behind [to/from]-[csv/tsv] commands
Following on from #943 where a separator arg was added to `from-csv`.

I've now realized that `from-tsv` is just a specific case of `from-csv` (tab separated values).
I've also noticed that passing `--separator "\t"` to `from-csv` trips my uninspired check for a single character (it sees two characters, '\' and 't').

I'd like to propose having `from-tsv` and `from-csv --separator "\t"` do the same thing, and I think we can deduplicate some code while we're at it. And I'm planning to make a pull request with these changes :)

1. [from/to]_tsv can call the same function as [from/tp]_csv, this has already been started with `from_structured_data`. Only, I think the scope of this module would be more clear if we named it `from_delimited_data`. This will use `csv::ReaderBuilder` and `csv::WriterBuilder` with different delimiters.

2. the parsing of the separator arg of from_csv and to_csv should be improved. I struggled to get a character out of nu from the command line and ended up checking a `string.to_chars()` has length 1. I can check if the string is "\t" first and return '\t' early for now but I worry that's symptomatic of a deeper issue? Are there plans for supporting characters from the command line? I'd be interested in the why/why not and how?

Also a side note: maybe --delimiter should be added as an alias for --separator?

### PR DESCRIPTION
**Pull request by drmason13**
fixes #969, admittedly without a --delimiter alias

moves from_structured_data.rs to from_delimited_data.rs to better
identify its scope and adds to_delimited_data.rs. Now csv and tsv both
use the same code, tsv passes in a fixed '\t' argument where csv passes
in the value of --separator

### CODE REVIEW

File: tests/filters_test.rs [hunk 1]
```diff
@@ -267,6 +296,16 @@ fn can_convert_table_to_tsv_text_and_from_tsv_text_back_into_table() {
     assert_eq!(actual, "SPAIN");
 }
 
+#[test]
+fn can_convert_table_to_tsv_text_and_from_tsv_text_back_into_table_using_csv_separator() {
+    let actual = nu!(
+        cwd: "tests/fixtures/formats",
+        r"open caco3_plastics.csv | to-tsv | from-csv --separator '\t' | first 1 | get origin | echo $it"
```
Comment (andrasio) [inline tests/filters_test.rs]: Shouldn't this be caco3_plastics.tsv ?

Comment (andrasio) [reply tests/filters_test.rs]: Nevermind :)

Comment (drmason13) [reply tests/filters_test.rs]: ah yes! it should, good spot

Comment (drmason13) [reply tests/filters_test.rs]: let me just force push over this, if that's ok with you?

Comment (andrasio) [reply tests/filters_test.rs]: sure, thanks!

Comment (andrasio) [reply tests/filters_test.rs]: Although I think this test is correct. You are opening a csv file and getting a table, converting to tsv and using the expected `from-csv` with the separator wanted in this case.

This is correct.

Comment (drmason13) [reply tests/filters_test.rs]: it's slightly confusing... but yes, it wouldn't matter what format the file was as long as it reads as a table, then we make tsv text (so we _add_ tabs) and then we parse that as csv separated by tabs...

I think this does the job of demonstrating that `tsv` and `csv --separator '\t'` are interoperable :D

**Pull request merged**